simulator/modules/core/perf_instr.h

```diff
@@ -22,6 +22,23 @@ class PerfInstr : public FuncInstr
     BPInterface get_bp_upd() const { return BPInterface( this->get_PC(), this->is_jump_taken(), this->get_new_PC()); }
 
     auto is_bypassible() const { return !this->is_conditional_move() && !this->is_accumulating_instr(); }
+    
+    auto is_complex_arithmetic() const { return this->is_divmult(); }
+    auto is_mem_stage_required() const { return this->is_load()  ||
+                                                this->is_store() ||
+                                                this->is_jump()  ||
+                                                this->is_explicit_trap()  ||
+                                                this->is_special(); }
+    
+    auto get_cycles_till_writeback() const
+    {
+        if ( is_complex_arithmetic())
+            return 3_Cl;
+        else if ( is_mem_stage_required())
+            return 2_Cl;
```

Comment:
Could you please make it configurable with our infrastructure?

Reply:
Yes, we need to have different options for latencies (maybe not in that commit, but in the next one). 'Complex' arithmetics may be 10 cycles, or data cache hit is 4 cycles, etc — let's start with two knobs.